Wangscape/noise/ModuleGroupContainer.h

```diff
@@ -17,8 +18,8 @@ class ModuleGroupContainer
 public:
     explicit ModuleGroupContainer(std::string module_group_description_);
     virtual ~ModuleGroupContainer() = default;
-    void addSpecificModuleGroup(T key, std::string filename);
-    void tryAddDefaultModuleGroup(T key, boost::optional<std::string> filename);
+    void addSpecificModuleGroup(T key, std::string filename, std::mt19937& rng);
+    void tryAddDefaultModuleGroup(T key, boost::optional<std::string> filename, std::mt19937& rng);
```

Comment:
I don't like this interface. I see two problems here:
* as this is `ModuleGroupContainer` and it has `*add*ModuleGroup` methods, I'd expect that they accept already loaded `ModuleGroup`, not filename to it (but it's a different issue)
* as you're calling `setSeeds` after every `insert` and the issue you want to resolve is called "Call setSeeds() on border and combiner module groups **during initialisation**", the best solution in my opinion would be calling `setSeeds` in the `ModuleGroup`'s constructor. By doing so, you've got guarantee that it is always called. Good idea could be also passing seed to it, as that class doesn't need RNG after generating seed.

Do you agree?

Reply:
* I could rename them to `*load*ModuleGroup` or add an intermediate function, but there's no reason to keep a `ModuleGroup` object around after it's added to a `ModuleGroupContainer`. Duplicate module groups must be fully independent copies (so that they can be used together with different seeds), and libnoise `Module`s aren't copyable. This means a `ModuleGroup` must be loaded every time it's added to a `ModuleGroupContainer`.
* The `setSeeds()` call could be moved to `loadModuleGroup()`. I don't think it should be moved as far as the `ModuleGroup` constructor or the `EncodedModuleGroup::decode()` call.

Reply:
Looks good :).